cylc/flow/task_proxy.py

```diff
@@ -32,6 +32,79 @@
 from cylc.flow.wallclock import get_unix_time_from_time_string as str2time
 
 
+def generate_graph_children(tdef, point):
+    # Determine graph children of this task (for spawning).
+    graph_children = {}
+    for seq, dout in tdef.graph_children.items():
+        for output, downs in dout.items():
+            if output not in graph_children:
+                graph_children[output] = []
+            for name, trigger in downs:
+                child_point = trigger.get_child_point(point, seq)
+                is_abs = (trigger.offset_is_absolute or
+                          trigger.offset_is_from_icp)
+                if is_abs:
+                    if trigger.get_parent_point(point) != point:
+                        # If 'foo[^] => bar' only spawn off of '^'.
+                        continue
+                if seq.is_on_sequence(child_point):
+                    # E.g.: foo should trigger only on T06:
+                    #   PT6H = "waz"
+                    #   T06 = "waz[-PT6H] => foo"
+                    graph_children[output].append((name, child_point, is_abs))
+
+    if tdef.sequential:
+        # Add next-instance child.
+        nexts = []
+        for seq in tdef.sequences:
+            nxt = seq.get_next_point(point)
+            if nxt is not None:
+                # Within sequence bounds.
+                nexts.append(nxt)
+        if nexts:
+            if TASK_OUTPUT_SUCCEEDED not in graph_children:
+                graph_children[TASK_OUTPUT_SUCCEEDED] = []
+            graph_children[TASK_OUTPUT_SUCCEEDED].append(
+                (tdef.name, min(nexts), False))
+
+    return graph_children
+
+
+def generate_graph_parents(tdef, point):
+    # Determine graph parents of this task.
+    graph_parents = {}
+    for seq, ups in tdef.graph_parents.items():
+        graph_parents[seq] = []
+        for name, trigger in ups:
+            parent_point = trigger.get_parent_point(point)
+            is_abs = (trigger.offset_is_absolute or
+                      trigger.offset_is_from_icp)
+            if is_abs:
+                if parent_point != point:
+                    # If 'foo[^] => bar' only spawn off of '^'.
+                    continue
+            if seq.is_on_sequence(parent_point):
+                # E.g.: foo should trigger only on T06:
+                #   PT6H = "waz"
+                #   T06 = "waz[-PT6H] => foo"
+                graph_parents[seq].append((name, parent_point, is_abs))
```

Comment:
The comments here relate to spawning and triggering, which isn't appropriate to graph parents.  But also, have you convinced yourself this code works in the parent case (for the child case, I remember this stuff being a bit brain-twisting...)

Reply:
Yeah, because the appropriate history remains in the graph window when you prune out path nodes.
Another aspect working; it doesn't go back before the initial cycle point.

Reply:
Without it, the oldest task is in the active pool.